src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/KafkaService.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.api.rest.services
+
+import org.apache.kafka.clients.consumer.{ConsumerConfig, KafkaConsumer}
+import org.apache.kafka.common.TopicPartition
+import org.springframework.stereotype.Service
+import za.co.absa.hyperdrive.trigger.api.rest.services.KafkaServiceImpl.{BeginningOffsets, EndOffsets, OffsetFunction}
+import za.co.absa.hyperdrive.trigger.configuration.application.{GeneralConfig, KafkaConfig}
+
+import java.util.Properties
+import java.util.concurrent.ConcurrentHashMap
+import javax.inject.Inject
+import scala.collection.JavaConverters._
+
+trait KafkaService {
+  def getBeginningOffsets(topic: String, consumerProperties: Properties): Map[Int, Long]
+  def getEndOffsets(topic: String, consumerProperties: Properties): Map[Int, Long]
+}
+
+@Service
+class KafkaServiceImpl @Inject() (kafkaConfig: KafkaConfig, generalConfig: GeneralConfig) extends KafkaService {
+
+  private val kafkaConsumersCache = new ConcurrentHashMap[Properties, KafkaConsumer[String, String]]()
```

Comment:
Using just hash map may lead to out of mem issue.
If you update a workflow and change kafka sensor to time, consumer will stay in memory (also if you just change kafka sensor conf). 
If all instances go down except one, that instance will keep in mem consumers for all kafka sensors.

I think at least some cache with time expiration is needed or just create/destroy consumer every time